### PR TITLE
Bug 1434003 - allow task template to specify timestamps

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

File: test/taskcreator_test.js [hunk 1]
```diff
@@ -92,6 +109,20 @@ suite('TaskCreator', function() {
     });
   });   
 
+  test('firing a real task that sets its own task times works', async function() {
+    let hook = _.cloneDeep(defaultHook);
+    hook.task.created = {$fromNow: '0 seconds'};
```
Comment (alexandrasp) [inline test/taskcreator_test.js]: more JSON-e : r+++

**Comment by alexandrasp** [comment]
probably this commit (and others) will conflict with trigger by pulse message branch :}

File: test/taskcreator_test.js [hunk 2]
```diff
@@ -29,40 +30,56 @@ suite('TaskCreator', function() {
...
+        name:           'Unit testing task',
+        description:    'Task created during unit tests',
+        owner:          'amiyaguchi@mozilla.com',
+        source:         'http://github.com/',
+      },
+      tags: {
+        purpose:        'taskcluster-testing',
+      },
+    },
+    bindings:           [],
```
[the remark this answers is not in the document]
Comment (djmitche) [reply test/taskcreator_test.js]: Yeah, I'm just supplying the default value here.

**Comment by djmitche** [comment]
Hopefully it won't conflict too badly..

**Pull request merged**